### PR TITLE
chore: Upgrade to Go 1.21 (rebase developer image from ubi/go-toolset to ubi-minimal)

### PR DESCRIPTION
**Pull request by ckadner**
#### Motivation

In preparation of the v0.12.0 release of ModelMesh Serving, we need to upgrade from Go 1.20 to Go 1.21 as required by KServe v0.12.0.

Since there will not be a UBI `go-toolset` image for Go `1.21` until  2Q 2024, we need to "rebase" the Developer image from `ubi8/go-toolset:1.20` to `ubi8/ubi-minimal:8.9` and install all required tools that would otherwise be provided "out of the box".

#### Modifications

First step, "rebase" the Developer image from `ubi8/go-toolset:1.20` to `ubi8/ubi-minimal:8.9` and install Go 1.20.10 along with all required tools. Keep with Go 1.20.10 at first to ensure there are no regressions.

#### Result

TBD - Work in progress

### CODE REVIEW

**Comment by ckadner** [comment]
@spolti @rafvasq 

I am leaving this PR as is:
- rebase dev image (`ubi/go-toolset:1:20` => `ubi8/ubi-minimal:8.9`)
- update from Go `1.20` to Go `1.21`

I am not yet:
- upgrading KServe from `v0.11.2` to `v0.12.0-rc1`
- not switching from `ubi8` to `ubi9`

In order to keep the changes to a minimum and make it easier to back track should there be unforeseen side effects. 

I will create separate PRs for the UBI and KServe upgrades, as the UBI upgrade spans several repositories and the KServe update has a long tail of related changes that require a bit more work.

File: go.mod [hunk 1]
```diff
@@ -129,15 +126,17 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+// when adding/remove replace, remove the following block of indirect dependencies
+// and run `go mod tidy -compat=1.20` (based on go version above)
+require cloud.google.com/go v0.110.2 // indirect
```
Comment (spolti) [inline go.mod:131]: why moving it outside the first required section?

Comment (ckadner) [reply go.mod]: so weird. no idea how that happened. thanks for catching that.

Comment (ckadner) [reply go.mod]: I cleaned it up :-)

**Review by rafvasq — APPROVED** [review]

**Comment by ckadner** [comment]
Thanks @spolti @rafvasq for your reviews

**Pull request merged**